src/opera_disp_tms/generate_sw_disp_tile.py

```diff
@@ -117,18 +117,23 @@ def add_granule_data_to_array(
     Returns:
         The updated short wavelength cumulative displacement array and the secondary date of the granule
     """
-    granule_dataarray = open_opera_disp_granule(granule.s3_uri, 'short_wavelength_displacement')
+    datasets = ['short_wavelength_displacement', 'connected_component_labels']
+    granule_xr = open_opera_disp_granule(granule.s3_uri, datasets)
 
-    same_ref_date = round_to_day(granule_dataarray.attrs['reference_date']) == round_to_day(frame.reference_date)
+    same_ref_date = round_to_day(granule_xr.attrs['reference_date']) == round_to_day(frame.reference_date)
     if not same_ref_date:
         raise NotImplementedError('Granule reference date does not match metadata tile, this is not supported.')
 
-    granule_dataarray = granule_dataarray.rio.reproject('EPSG:3857', transform=geotransform, shape=frame_map.shape)
+    bbox = create_buffered_bbox(geotransform.to_gdal(), frame_map.shape, 120)  # EPSG:3857 is in meters
+    granule_xr = granule_xr.rio.clip_box(*bbox, crs='EPSG:3857')
+    conn_comp_mask = granule_xr['connected_component_labels'] == 0
+    sw_cumul_disp_xr = granule_xr['short_wavelength_displacement'].where(conn_comp_mask, np.nan)
```

Comment:
The `.where` condition is backwards, see https://docs.xarray.dev/en/stable/generated/xarray.Dataset.where.html#xarray.Dataset.where

```suggestion
    valid_data_mask = granule_xr['connected_component_labels'] > 0
    sw_cumul_disp_xr = granule_xr['short_wavelength_displacement'].where(valid_data_mask, np.nan)
```

Reply:
![Screenshot from 2024-10-11 14-23-03](https://github.com/user-attachments/assets/fb68ce80-fcaa-4347-9b4b-369a9527d204)

Reply:
Looks great once the mask is inverted!

https://asj-dev.s3.amazonaws.com/opera-disp/mask/asc/openlayers.html

https://asj-dev.s3.amazonaws.com/opera-disp/mask/desc/openlayers.html

Reply:
I don't know if this makes it harder or easier to read, but since the default for xr.where is a NAN, we could also write this as :
`sw_cumul_disp_xr = granule_xr['short_wavelength_displacement'].where(valid_data_mask)`
OR 
`sw_cumul_disp_xr = granule_xr['short_wavelength_displacement'].where(granule_xr['connected_component_labels'] > 0)`

Reply:
Ope, thanks for the catch @asjohnston-asf!

Also thanks for the style suggestion @jacquelynsmale. However, I think I prefer it as-is. I like being explicit about what value we're setting nodata to for readability, and in case xarray ever changes its behavior.

This decision makes the `where` line longer than 120 characters, and I think the current approach of creating a `valid_data_mask` is the most readable in this case.